docs/PasswordManagement.md

```diff
@@ -0,0 +1,69 @@
+# Password Management
+
+- [Global kubernetes secret object](#global-kubernetes-secret-object)
+- [Default behavior of global kubernetes secret object](#default-behavior-of-global-kubernetes-secret-object)
+  - [Splunk Secret Tokens in the global secret object](#splunk-secret-tokens-in-the-global-secret-object)
+    - [HEC Token](#hec-token)
+    - [Default administrator password](#default-administrator-password)
+    - [Pass4symmkey](#pass4symmkey)
+    - [IDXC Pass4symmkey](#idxc-pass4symmkey)
+    - [SHC Pass4symmkey](#shc-pass4symmkey)
+- [Information for Splunk Enterprise administrator](#information-for-splunk-enterprise-administrator)
+- [Secrets on Docker Splunk](#secrets-on-docker-splunk)
+
+## Global kubernetes secret object
+A global kubernetes secret object acts as a common source of splunk secret tokens for all Splunk Enterprise CRs in a kubernetes namespace. It's name follows the format "splunk-<namespace>-secret" where <namespace> represents the namespace we are operating in. The contents of this object are volume mounted on all the pods within a kubernetes namespace. 
+
+This approach:
+  - Eliminates any mismatch between operator-generated secrets and admin provided secrets as all secrets are synced into a common object.
+  - Allows for dynamic adoption and modification of secrets.
+
+## Default behavior of global kubernetes secret object
+
+For a given namespace upon the creation of the first Splunk Enterprise CR, the operator checks for the existence of a global kubernetes secret object:
+
+- If the object does not exist,
+    - Creates the object
+    - Auto-generates(except for S3 specific tokens) & stores all the Splunk secret tokens into the global kubernetes secret object
+- If the object exists,
+    - Reads the configured Splunk secret tokens in the global secret object
+    - Auto-generates(except for S3 specific tokens) & stores only the missing Splunk secret tokens into the global secret object
+
+Note: Before the creation of any Splunk deployments in the kubernetes namespace, the admin can create a global kubernetes secret object with the tokens mentioned below so that the operator uses these values to deploy.
+
+### Splunk Secret Tokens in the global secret object
+The configurable Splunk Secret Tokens include:
+
+#### HEC Token
+**Key String in global kubernetes secret object**: `hec_token`  
+**Description**: hec_token is used to authenticate clients sending data into Splunk Enterprise via HTTP connections.
+
+#### Default administrator password
+**Key String in global kubernetes secret object**: `password`  
+**Description**: password refers to the default administrator password for Splunk. 
```

Comment:
Wondering if we need to mention that, Splunk local authentication is a must for admin, and external authentication like LDAP/SAML can not be used?

Reply:
Can't we use LDAP/SAML on top of the admin password configured?

Reply:
Done